Bring up a Test262 EWS queue on Ventura

robert-jenner opened the pull request:
#### 6442231a3eb1961f486c400589cd6c99926e28b7
<pre>
Bring up a Test262 EWS queue on Ventura
<a href="https://bugs.webkit.org/show_bug.cgi?id=255967">https://bugs.webkit.org/show_bug.cgi?id=255967</a>
rdar://problem/107203107

Reviewed by NOBODY (OOPS!).

An attempt to modify our EWS config files to add a test262 queue.

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories.py:
(JSCTest262BuildAndTestsFactory):
(JSCTest262BuildAndTestsFactory.__init__):
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(RunJSCTestsWithoutChange.evaluateCommand):
(RunTest262Tests):
(RunTest262TestsWithoutChange):
(RunTest262TestsWithoutChange.evaluateCommand):
(AnalyzeTest262TestsResults):
(AnalyzeTest262TestsResults.start):
(AnalyzeTest262TestsResults.retry_build):
(AnalyzeTest262TestsResults.report_failure):
(AnalyzeTest262TestsResults.send_email_for_flaky_failure):
(AnalyzeTest262TestsResults.send_email_for_pre_existing_failure):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6442231a3eb1961f486c400589cd6c99926e28b7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4761 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4879 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5042 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6267 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4889 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4753 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5032 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4847 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5124 "1 flakes 106 failures") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4839 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4917 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4256 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6278 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2405 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4251 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9222 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4267 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4325 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5894 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4734 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3851 "Passed tests") | | 
| [❌ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/4720 "Failed EWS unit tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4247 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8303 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4604 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3625099b8355ebdc3cf366f49db331840f4a6ce2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8f0290a93d29611ebefcfed6e2c6248b32cdf63e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3a7a2737aa36a27f7babf2d24400a35df7dea77e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1d646d3ee55025ca49ffd485be84d720a0168c49. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e9b08ad1553f6bb7ce439097b138ea27e9a5a1df. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4f592bf2c68facb1e36eb4c3f63c3f61c9b72516. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3c8132ec59ac3cf3c21372e4e11c5ed68e1b5a80. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9be5842ec4655ea2970121755b15101aaab01732. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

robert-jenner committed:
Bring up a Test262 EWS queue on Ventura
https://bugs.webkit.org/show_bug.cgi?id=255967
rdar://problem/107203107

Reviewed by NOBODY (OOPS!).

An attempt to modify our EWS config files to add a test262 queue.

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories.py:
(JSCTest262BuildAndTestsFactory):
(JSCTest262BuildAndTestsFactory.__init__):
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(RunJSCTestsWithoutChange.evaluateCommand):
(RunTest262Tests):
(RunTest262TestsWithoutChange):
(RunTest262TestsWithoutChange.evaluateCommand):
(AnalyzeTest262TestsResults):
(AnalyzeTest262TestsResults.start):
(AnalyzeTest262TestsResults.retry_build):
(AnalyzeTest262TestsResults.report_failure):
(AnalyzeTest262TestsResults.send_email_for_flaky_failure):
(AnalyzeTest262TestsResults.send_email_for_pre_existing_failure):

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6442231a3eb1961f486c400589cd6c99926e28b7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13247

JonWBedard commented on Tools/CISupport/ews-build/factories.py:
```diff
@@ -194,6 +194,16 @@ def __init__(self, platform, configuration='release', architectures=None, remote
+        self.addStep(KillOldProcesses())
+        self.addStep(ValidateChange(addURLs=False))
+        self.addStep(CompileJSC(skipUpload=True))
+        if runTests.lower() == 'true':
```
Why is this a string instead of a boolean?

robert-jenner replied:
It's basically just a copy paste of `class JSCBuildAndTestsFactory(Factory):` a few lines above this one. The only thing I modified was `self.addStep(RunTest262Tests())` at the end instead of `self.addStep(RunJavaScriptCoreTests())` otherwise I didn't modify it any differently from what was already there.

JonWBedard commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3229,6 +3228,157 @@ def send_email_for_pre_existing_failure(self, test_name):
+        self._addToLog('stderr', '\nFlaky Tests on clean tree: {}'.format(', '.join(clean_tree_flaky_stress_failures)))
+
+        if (not stress_failures_with_change) and (not binary_failures_with_change):
+            # If we've made it here, then jsc-tests and re-run-jsc-tests failed, which means
```
This comment doesn't seem correct for this queue.

robert-jenner replied:
I completely agree. But I wasn't certain how to specifically deal with removing all of the JSC Stress test failure lines. Basically everything under `class AnalyzeTest262TestsResults(buildstep.BuildStep, AddToLogMixin):` needs a lot of work, and I'm not entirely certain how to proceed.

robert-jenner commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3229,6 +3228,157 @@ def send_email_for_pre_existing_failure(self, test_name):
             print('Error in sending email for pre-existing failure: {}'.format(e))
 
 
+class RunTest262Tests(TestWithFailureCount):
```
It was lifted from Line 640 in `Tools/CISupport/build-webkit-org/steps.py`
